src/OrleansRuntime/GrainTypeManager/GrainTypeData.cs

```diff
@@ -151,5 +154,11 @@ internal static PlacementStrategy GetPlacementStrategy(Type grainClass, Placemen
 
             return predicate;
         }
+
+        private ParameterInfo[] GetConstructorParameters()
+        {
+            var constructor = this.Type.GetConstructors(BindingFlags.Public | BindingFlags.Instance).FirstOrDefault();
+            return constructor?.GetParameters() ?? EmptyParameters;
+        }
```

Comment:
@ReubenBond, Of course we can move it out, but that seems arbitrary.  Can you provide a technical reason why it shouldn't be kept with all of the other grain type static data?  I'm not opposed to moving it elsewhere, but not on the basis of personal whims.  Please provide technical reasons for why this approach is bad/wrong and alternatives with their advantages.

What we need is a common way to provide constructed types wired up with specific metadata to a grains construction.  These types need to be able to take part in the grain lifecycle.  Originally this was done by passing the constructed objects as arguments during construction, but that was not supported by the asp.net controller pattern, so we moved to matching up items to parameters in the constructor (like in this PR), which necessitates the need for a well defined constructor selection process, which fortunately the DI specification we've adapted supports.  Where we cache the parameters is an internal detail and is not that important, but I am opposed to adding new static data sets without reason.